### PR TITLE
PDF and Excel

### PR DESCRIPTION
**Pull request by zaucker**
- Factor out export data collection
- Refactor Excel generation
- Add PDF generation

More/nicer content for both exports will follow in separate PR

### CODE REVIEW

File: lib/Agrammon/OutputFormatter/PDF.pm6 [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    if $signal {
+        note "$pdf-prog killed for $source-file, signal=$signal, reason=$reason";
+        die X::Agrammon::OutputFormatter::PDF::Killed.new: :$reason;
+    }
+
+    # read content of PDF file created
+    my $pdf = $pdf-file.slurp(:bin);
+
+    # cleanup if successful
+    unlink $source-file, $pdf-file, $aux-file, $log-file;
```
[the remark this answers is not in the document]
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/PDF.pm6:80]: Yes

File: lib/Agrammon/OutputFormatter/PDF.pm6 [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    my $signal;
+    my $reason = 'Unknown';
+
+    my $proc = Proc::Async.new: :w, $pdf-prog, "--output-directory=$temp-dir", '--safer', '--no-shell-escape', '--', $source-file, ‘-’;
+    react {
+        # just ignore them
+        whenever $proc.stdout.lines {
+        }
+        whenever $proc.stderr {
+        }
```
[the remark this answers is not in the document]
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/PDF.pm6:50]: Well, I don't want them in the syslog. All I need is in the `.log` file. So I should probably fix the comment?

File: lib/Agrammon/OutputFormatter/CollectData.pm6 [hunk 3]
```diff
@@ -0,0 +1,125 @@
+use v6;
+use Agrammon::Model;
+use Agrammon::Outputs;
+use Agrammon::Outputs::FilterGroupCollection;
+
+# TODO: make output match current Agrammon Excel export
```
[the remark this answers is not in the document]
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/CollectData.pm6:6]: Yes, both Excel and PDF don't have all the infos and format yet.

File: frontend/source/class/agrammon/module/output/Reports.js [hunk 4]
```diff
@@ -214,55 +214,48 @@ qx.Class.define('agrammon.module.output.Reports', {
...
+            }
+            document.body.appendChild(form);
 
-        // 	// Response parsed according to the server's
-        // 	// response content type, e.g. JSON
-        // 	var response = req.getResponse();
-        // 	this.debug('response='+response);
-        // }, this);
-        // req.send();
+            // status=1 open's in new tab in Chrome at least
```
[the remark this answers is not in the document]
Comment (zaucker) [reply frontend/source/class/agrammon/module/output/Reports.js:249]: Fixed

File: lib/Agrammon/OutputFormatter/CollectData.pm6 [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    Agrammon::Outputs $outputs, Agrammon::Inputs $inputs, $reports,
+    Str $language, $prints,
+    Bool $include-filters, Bool $all-filters
+) is export {
+    warn '**** collect-data() not yet completely implemented';
+
+    # TODO: add missing inputs
+    my @inputs;
+    my @annotated-inputs = $model.annotate-inputs($inputs);
+    for @annotated-inputs -> $ai {
```
[the remark this answers is not in the document]
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/CollectData.pm6:18]: Did 2.

File: lib/Agrammon/OutputFormatter/PDF.pm6 (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnthn) [reply lib/Agrammon/OutputFormatter/PDF.pm6:50]: Ah, if that's what you actually want then yeah.

File: lib/Agrammon/OutputFormatter/PDF.pm6 [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+         whenever $proc.start {
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/PDF.pm6:51]: Space too much indentation here

**Review by jnthn — APPROVED** [review]
One tiny thing spotted, but otherwise seems fine.

**Comment by zaucker** [comment]
Merging, will fix GitHub problem separately.

**Pull request merged**